### PR TITLE
Molotov prototype for testing

### PR DESCRIPTION
**Pull request by andrew-raphael-lukasik**
Hi,
I made an prototype for molotov cocktails. It's not tested well yet (standalone bots only), so I have no idea if its network replication code even works. I'm suggesting an PR here so you can easily play test it, give feedback maybe etc. and trash talk it.

[Preview images](https://imgur.com/a/5Yi1hFG)

I'm not terribly sure if pulling this on Master branch is a good idea necessarily. Dedicated branch might be more fitting

### CODE REVIEW

**Comment by andrew-raphael-lukasik** [comment]
https://www.youtube.com/watch?v=4s2Obr5WpBk
Finally figured out how to add fpp model without errors. Animations are coming next

File: DH_Effects/Classes/DH_MolotovMark.uc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	DrawScale=0.5
+	bGameRelevant=true
+	PushBack=24
+	LifeSpan=30
+	FOV=1
+	MaxTraceDistance=60
+	bProjectBSP=true
+	bProjectTerrain=true
+	bProjectStaticMesh=true
+	bProjectActor=true
```
Comment (shiftycats) [inline DH_Effects/Classes/DH_MolotovMark.uc]: `bProjectActor` should be set `false`. The projection is frozen in space; if the actor moves, you will see the blast mark sliding over it.

File: DH_Weapons/Classes/DH_MolotovProjectile.uc [hunk 2]
```diff
@@ -0,0 +1,644 @@
...
+
+    if( Role==ROLE_Authority )
+    {
+        MakeNoise( 1.0 );
+        if( Level.NetMode!=NM_DedicatedServer )
+        {
+            PlaySound( ExplosionSound[Rand(3)] ,, 5.0 ,, ExplosionSoundRadius , 1.0 , true );
+        }
+        
+        if( EffectIsRelevant(Location,false) )
```
Comment (shiftycats) [inline DH_Weapons/Classes/DH_MolotovProjectile.uc]: `Role == ROLE_Authority` and `EffectIsRelevant()` are mutually exclusive in multiplayer. Effects are only relevant on the client, while the the first check ensures that things are executed on the server only (the `BlowUp()` function is never called on the client either).

Comment (shiftycats) [reply DH_Weapons/Classes/DH_MolotovProjectile.uc]: Fixed in commit ebb83f57a1c2f5f2208f013182f4b5c3ee9dd868.

File: DH_Effects/Classes/DH_MolotovMark.uc (hunk 1 above)
Comment (shiftycats) [reply DH_Effects/Classes/DH_MolotovMark.uc]: Fixed in commit 857987d72bdaaae70274c307dd6c8b0b05d2f1e2.

**Pull request merged**